libs/core/src/lib/resizable-card-layout/resizable-card-layout.component.ts

```diff
@@ -0,0 +1,497 @@
+            resizeCardItem.resized
+                .pipe(takeUntil(this._destroy$))
+                .subscribe((event: ResizedEvent) => this.cardResizeComplete(event));
+
+            // listen for mini-header height event of card item
+            resizeCardItem.miniHeaderReached.pipe(takeUntil(this._destroy$)).subscribe((event: ResizedEvent) => {
+                this.miniHeaderReached.emit(event);
+            });
+
+            // listen for mini-content height event of card item
+            resizeCardItem.miniContentReached.pipe(takeUntil(this._destroy$)).subscribe((event: ResizedEvent) => {
+                this.miniContentReached.emit(event);
+            });
+
+            // listen for step-change event of card item
+            resizeCardItem.stepChange.pipe(takeUntil(this._destroy$)).subscribe((event: ResizedEvent) => {
+                this.stepChange.emit(event);
+            });
+        });
+    }
+
+    /** @hidden */
+    private _accessibilitySetup(): void {
+        this._keyboardEventsManager = new FocusKeyManager(this.resizeCardItems).withWrap();
+    }
+
+    /** Emit layoutChange event with updated card dimensions */
+    private _emitLayoutChange(): void {
+        const latestCardConfig: ResizableCardLayoutConfig = [];
+        this.resizeCardItems.forEach((card) => {
+            const cardConfig: ResizableCardItemConfig = {};
+            cardConfig.cardWidthColSpan = card.cardWidthColSpan;
+            cardConfig.cardHeightRowSpan = card.cardHeightRowSpan;
+            cardConfig.rank = card.rank;
+            latestCardConfig.push(cardConfig);
+        });
+        this.layoutChange.emit(latestCardConfig);
+    }
+
+    /** @hidden Sets number of column in layout, based on LayoutSize passed */
+    private _setLayoutColumns(layoutSize: LayoutSize): void {
+        switch (layoutSize) {
+            case 'sm':
+                this._columns = 1;
+                this._paddingLeft = 8;
+                this.layoutWidth = this._columns * HorizontalResizeStep + 2 * this._paddingLeft;
+                break;
+            case 'md':
+                this._columns = 2;
+                this._paddingLeft = 16;
+                this.layoutWidth = this._columns * HorizontalResizeStep + 2 * this._paddingLeft + gap;
+                break;
+            case 'lg':
+                this._columns = 3;
+                this._paddingLeft = 16;
+                this.layoutWidth = this._columns * HorizontalResizeStep + 2 * this._paddingLeft + 2 * gap;
+                break;
+            case 'xl':
+                this._columns = 4;
+                this._paddingLeft = 48;
+                this.layoutWidth = this._columns * HorizontalResizeStep + 2 * this._paddingLeft + 3 * gap;
+                break;
+            default:
+                this._columns = 4;
+                this._paddingLeft = 48;
+                this.layoutWidth = this._columns * HorizontalResizeStep + 2 * this._paddingLeft + 3 * gap;
+        }
```

Comment:
```suggestion
        const layoutWidthDefault = this._columns * HorizontalResizeStep + 2 * this._paddingLeft;
        switch (layoutSize) {
            case 'sm':
                this._columns = 1;
                this._paddingLeft = 8;
                this.layoutWidth = layoutWidthDefault;
                break;
            case 'md':
                this._columns = 2;
                this._paddingLeft = 16;
                this.layoutWidth = layoutWidthDefault + gap;
                break;
            case 'lg':
                this._columns = 3;
                this._paddingLeft = 16;
                this.layoutWidth = layoutWidthDefault + 2 * gap;
                break;
            case 'xl':
                this._columns = 4;
                this._paddingLeft = 48;
                this.layoutWidth = layoutWidthDefault + 3 * gap;
                break;
            default:
                this._columns = 4;
                this._paddingLeft = 48;
                this.layoutWidth = layoutWidthDefault + 3 * gap;
        }
```

Reply:
fixed.
have to keep `const layoutWidthDefault = this._columns * HorizontalResizeStep + 2 * this._paddingLeft;` inside case statement; as this._columns and this._paddingLeft; will not be available outside.